.github/workflows/docker-gdal.yml

```diff
@@ -25,8 +25,8 @@ jobs:
           - "ghcr.io/osgeo/gdal:ubuntu-small-3.8.5" # python 3.10.12
           - "ghcr.io/osgeo/gdal:ubuntu-small-3.7.3" # python 3.10.12
           - "ghcr.io/osgeo/gdal:ubuntu-small-3.6.4" # python 3.10.6
-          - "osgeo/gdal:ubuntu-small-3.5.3" # python 3.8.10
-          - "osgeo/gdal:ubuntu-small-3.4.3" # python 3.8.10
+          - "osgeo/gdal:ubuntu-small-3.5.3" # python 3.8.10?
+          - "osgeo/gdal:ubuntu-small-3.4.3" # python 3.8.10?
```

Comment:
I just pulled `osgeo/gdal:ubuntu-small-3.4.3` and `osgeo/gdal:ubuntu-small-3.5.3` locally and confirmed they have 3.8.10.

We could use [uv's python management](https://github.com/astral-sh/uv?tab=readme-ov-file#python-management) to bootstrap Python 3.9 instead of using the system one from GDAL Docker image for these tests.

Reply:
Yeah, I just added the question mark so that I could comment on those lines ;)

I was also thinking doing something similar as we do in the release workflow using uv, but then indeed installing another python version.

Another option would be to use `apt` to install another python version from the deadsnakes ppa (in other projects we also use that to install a non-system-supported python version in those cases).

Given we already started to use uv, will first give that a go.

Reply:
One thing I don't understand from the current failure on the docker build is why `uv` was trying to use Python 3.13.0rc2 for those older images ..

Reply:
Maybe because it noticed from pyproject.toml that the available python 3.8 would not work for this project?

Reply:
OK, @brendan-ward I tried both approaches (for the gdal 3.4 vs 3.5 images), and when using uv to get python 3.9, we still get the same error (that I don't really understand) about "clang" when building pyogrio (for some reason it tries to use clang for building instead of gcc? some consequence of how the python from uv is built?)

When the python is installed with the deadsnakes ppa, it also gets picked up automatically by uv but then building goes fine

Reply:
(updated to have both use the ppa approach)

Reply:
Thanks for trying it both ways!  Not sure why the uv version of python didn't work; it works properly when I try to build within these containers locally (Arm64 containers on MacOS M1 host).

Installing python via deadsnakes ppa is just fine too.